crates/swc_css_compat/src/compiler/nesting.rs

```diff
@@ -12,38 +12,49 @@ impl Compiler {
-            }) = sel
-            {
-                for c in children {
-                    if let PseudoClassSelectorChildren::ForgivingSelectorList(c) = c {
-                        let mut selectors = vec![];
-
-                        for sel in &mut c.children {
-                            match sel {
-                                ForgivingComplexSelector::ComplexSelector(sel) => {
-                                    selectors.push(sel.clone());
-                                }
-                                ForgivingComplexSelector::ListOfComponentValues(_) => {
-                                    // Abort
-                                    return;
+            match sel {
+                SubclassSelector::PseudoClass(PseudoClassSelector {
+                    children: Some(children),
+                    ..
+                }) => {
+                    for c in children {
+                        if let PseudoClassSelectorChildren::ForgivingSelectorList(c) = c {
+                            let mut selectors = vec![];
+
+                            for sel in &mut c.children {
+                                match sel {
+                                    ForgivingComplexSelector::ComplexSelector(sel) => {
+                                        selectors.push(sel.clone());
+                                    }
+                                    ForgivingComplexSelector::ListOfComponentValues(_) => {
+                                        // Abort
+                                        return;
+                                    }
                                 }
                             }
-                        }
 
-                        self.process_complex_selectors(prelude, &mut selectors);
+                            self.process_complex_selectors(prelude, &mut selectors);
 
-                        *c = ForgivingSelectorList {
-                            children: selectors
-                                .into_iter()
-                                .map(ForgivingComplexSelector::ComplexSelector)
-                                .collect(),
-                            ..*c
-                        };
+                            *c = ForgivingSelectorList {
+                                children: selectors
+                                    .into_iter()
+                                    .map(ForgivingComplexSelector::ComplexSelector)
+                                    .collect(),
+                                ..*c
+                            };
+                        }
                     }
                 }
+
+                SubclassSelector::PseudoElement(PseudoElementSelector { span, name, .. }) => {
+                    *sel = SubclassSelector::PseudoClass(PseudoClassSelector {
+                        span: *span,
+                        name: name.clone(),
+                        children: Default::default(),
+                    });
+                }
```

Comment:
Due this, I don't think it is valid solution, because your convert all pseudo elements to pseudo classes